cli/src/main/java/org/eclipse/hono/cli/AbstractCliClient.java

```diff
@@ -44,32 +45,79 @@
      */
     protected Context ctx;
     /**
-     * The Spring Boot profiles that are active.
+     * The client factory with methods to use.
      */
-    protected List<String> activeProfiles;
+    protected ClientConfig clientConfig;
+
+    protected ClientConfigProperties honoClientConfig;
 
     /**
-     * Sets the Spring environment.
-     * 
-     * @param env The environment.
-     * @throws NullPointerException if environment is {@code null}.
+     * Spring Shell inputReader.
      */
     @Autowired
-    public final void setActiveProfiles(final Environment env) {
-        Objects.requireNonNull(env);
-        activeProfiles = Arrays.asList(env.getActiveProfiles());
-    }
+    protected @Lazy InputReader inputReader;
 
+    /**
+     * Spring Shell output helper.
+     * <p>
+     * it can be passed to the method class instead to use the log (it's more stylish).
+     */
+    @Autowired
+    protected ShellHelper shellHelper;
+
+    /**
+     * Sets the default client configuration.
+     *
+     * @param honoClientConfig The Client Configuration class instance.
+     */
+    @Autowired
+    private void setClientConfigProperties(final ClientConfigProperties honoClientConfig) {
+        this.honoClientConfig = Objects.requireNonNull(honoClientConfig);
+    }
     /**
      * Sets the vert.x instance.
-     * 
+     *
      * @param vertx The vert.x instance.
      * @throws NullPointerException if vert.x is {@code null}.
      */
     @Autowired
-    public final void setVertx(final Vertx vertx) {
+    private void setVertx(final Vertx vertx) {
         this.vertx = Objects.requireNonNull(vertx);
         this.ctx = vertx.getOrCreateContext();
     }
 
+    /**
+     * Expose a class for the factory.
+     *
+     * @param config .
+     */
+    @Autowired
+    private void setClientProp(final ClientConfig config) {
+        this.clientConfig = config;
+    }
+
+    /**
+     * Check the availability of the factory and of the profile active.
+     * It allows the execution of sub commands of the application profile.
+     *
+     * @return Availability status.
+     */
+    protected Availability subCommandsAppAvailability() {
```

Comment:
FMPOV this belongs into the class implementing all the application related commands ...